gnocchi/rest/api.py

```diff
@@ -1484,20 +1482,32 @@ def post(self, metric_id, start=None, stop=None, aggregation='mean',
                 abort(400, "Invalid value for stop")
 
         try:
-            return {
-                str(metric.id): values
-                for metric, values in six.iteritems(
-                    pecan.request.storage.search_value(
-                        metrics, query, start, stop, aggregation,
-                        granularity
-                    )
-                )
-            }
+            predicate = storage.MeasureQuery(query)
         except storage.InvalidQuery as e:
             abort(400, six.text_type(e))
+
+        if granularity is not None:
+            granularity = [utils.to_timespan(g)
+                           for g in arg_to_list(granularity)]
+
+        results = {}
+
+        try:
+            for metric in metrics:
+                if granularity is None:
+                    granularity = [d.granularity
+                                   for d in metric.archive_policy.definition]
+                results[str(metric.id)] = []
+                for g in sorted(granularity):
+                    results[str(metric.id)].extend(
+                        pecan.request.storage.find_measure(
+                            metric, predicate, g, aggregation,
+                            start, stop))
```

Comment:
you lost concurrency in this patch. what's up?

Reply:
Yeah, I did not bother re-implementing it as it complicates the code.

I'm pretty sure nobody ever used that feature anyway. If you really want I can add a few lines to parallelize stuff again.

Reply:
well it's on by default...

i'd prefer we keep it. personal experience, threading in python works well as long as you know it's I/O.

Reply:
:(

Reply:
well, i mean if you can argue that there isn't much to make concurrent than that's valid but right your technically making this slower and i'm not sure 'no one uses it' is a valid reason.